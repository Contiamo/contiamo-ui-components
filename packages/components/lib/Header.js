"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var glamorous_1 = require("glamorous");
var constants_1 = require("./constants");
var Container = glamorous_1.default.header(function (_a) {
    var theme = _a.theme;
    return ({
        label: "header",
        display: "flex",
        justifyContent: "space-between",
        backgroundColor: theme.colors.white,
        height: constants_1.headerHeight,
        alignItems: "center",
        padding: "0 " + theme.spacing * 4 / 3 + "px",
        boxShadow: theme.shadows.card
    });
});
var Header = function (props) { return (React.createElement(Container, { key: props.id, css: props.css, className: props.className }, props.children)); };
exports.default = Header;
//# sourceMappingURL=Header.js.map