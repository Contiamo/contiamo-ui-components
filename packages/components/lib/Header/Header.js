"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var glamorous_1 = require("glamorous");
var utils_1 = require("@operational/utils");
var HeaderItem_1 = require("./HeaderItem");
exports.HeaderItem = HeaderItem_1.default;
var HeaderTitle_1 = require("./HeaderTitle");
exports.HeaderTitle = HeaderTitle_1.default;
var HeaderSeparator_1 = require("./HeaderSeparator");
exports.HeaderSeparator = HeaderSeparator_1.default;
var Container = glamorous_1.default.header(function (_a) {
    var theme = _a.theme, color = _a.color;
    var white = theme.colors.white;
    var backgroundColor = color ? utils_1.hexOrColor(color)(theme.colors[color] || white) : white;
    return {
        backgroundColor: backgroundColor,
        display: "flex",
        flex: "0 0 60px",
        height: 60,
        alignItems: "center",
        padding: theme.spacing / 2 + "px " + theme.spacing + "px",
        color: utils_1.readableTextColor(backgroundColor)(["black", "white"])
    };
});
var Header = function (props) { return (React.createElement(Container, { key: props.id, css: props.css, className: props.className, color: props.color }, props.children)); };
exports.default = Header;
//# sourceMappingURL=Header.js.map