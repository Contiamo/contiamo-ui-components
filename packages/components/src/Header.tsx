import * as React from "react"
import glamorous, { GlamorousComponent } from "glamorous"
import { Theme } from "@operational/theme"
import { hexOrColor, readableTextColor } from "@operational/utils"

export interface IProps {
  id?: string | number
  className?: string
  css?: any
  children: React.ReactNode
  theme?: Theme
  color?: string
}

const Container = glamorous.header(({ theme, color }: { theme: Theme; color?: string }): {} => {
  const { white } = theme.colors
  const backgroundColor = color ? hexOrColor(color)(theme.colors[color] || white) : white

  return {
    backgroundColor,
    display: "flex",
    flex: "0 0 60px",
    height: 60,
    alignItems: "center",
    padding: `${theme.spacing / 2}px ${theme.spacing}px`,
    color: readableTextColor(backgroundColor)(["black", "white"])
  }
})

const Header = (props: IProps) => (
  <Container key={props.id} css={props.css} className={props.className} color={props.color}>
    {props.children}
  </Container>
)

export default Header
