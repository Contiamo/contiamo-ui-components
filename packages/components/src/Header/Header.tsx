import * as React from "react"
import glamorous, { GlamorousComponent } from "glamorous"
import { Theme } from "@operational/theme"

import { WithTheme, Css, CssStatic } from "../types"

export interface Props {
  id?: string
  className?: string
  css?: Css
  children?: React.ReactNode
}

const Container = glamorous.header(({ theme }: WithTheme): CssStatic => ({
  label: "header",
  display: "flex",
  justifyContent: "space-between",
  backgroundColor: theme.colors.white,
  height: theme.box,
  alignItems: "center",
  padding: `0 ${theme.spacing * 1.5}px`,
  boxShadow: theme.shadows.card,
}))

const Header = (props: Props) => (
  <Container id={props.id} css={props.css} className={props.className}>
    {props.children}
  </Container>
)

export default Header
