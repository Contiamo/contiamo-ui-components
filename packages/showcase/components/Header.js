import * as React from "react"
import glamorous from "glamorous"
import { Header, Breadcrumb, Breadcrumbs } from "@operational/components"
import { operational } from "@operational/theme"
import Link from "next/link"

export default props => {
  const pathnameChunks = props.pathname.split("/").filter(s => s !== "")
  const chunk1 = pathnameChunks[0]
  const chunk2 = pathnameChunks[1]
  let currentPath = ""
  const breadcrumbs = pathnameChunks.map((chunk, index) => {
    currentPath += `/${chunk}`
    return {
      url: currentPath,
      label: props.pathmap[currentPath].query.title
    }
  })
  return (
    <Header
      css={{
        boxShadow: "0px 1px 2px #d3d1d1",
        justifyContent: "space-between",
        backgroundColor: "#fff",
        padding: "0 16px"
      }}
      color="#fff"
    >
      {
        <Breadcrumbs>
          {breadcrumbs.map((breadcrumb, index) => (
            <Breadcrumb key={index}>
              <Link key={index} href={breadcrumb.url}>
                <a>{breadcrumb.label}</a>
              </Link>
            </Breadcrumb>
          ))}
        </Breadcrumbs>
      }
      {props.note ? props.note : null}
    </Header>
  )
}
