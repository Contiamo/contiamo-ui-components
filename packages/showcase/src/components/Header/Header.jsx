// @flow
import React from "react"
import { Route } from "react-router-dom"

import { Header } from "contiamo-ui-components"

const style: {} = {
  boxShadow: "0px 1px 2px #d3d1d1",
  backgroundColor: "#fff"
},
  AppHeader = () =>
    <Header css={style} color="#fff">
      Contiamo UI
      <Route
        path="/components"
        render={() =>
          <div>
            &nbsp;/&nbsp;<strong>Components</strong>
          </div>}
      />
    </Header>

export default AppHeader
