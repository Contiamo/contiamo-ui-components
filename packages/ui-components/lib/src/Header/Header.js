"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var glamorous_1 = require("glamorous");
var contiamo_ui_utils_1 = require("contiamo-ui-utils");
var HeaderItem_1 = require("./Item/HeaderItem");
exports.HeaderItem = HeaderItem_1.default;
var HeaderTitle_1 = require("./Title/HeaderTitle");
exports.HeaderTitle = HeaderTitle_1.default;
var HeaderSeparator_1 = require("./Separator/HeaderSeparator");
exports.HeaderSeparator = HeaderSeparator_1.default;
var Container = glamorous_1.default.header(function (_a) {
    var theme = _a.theme, color = _a.color;
    var backgroundColor = color ? contiamo_ui_utils_1.hexOrColor(color)(theme.colors.palette[color] || "white") : "white";
    return {
        backgroundColor: backgroundColor,
        display: "flex",
        flex: "0 0 50px",
        height: 50,
        alignItems: "center",
        padding: theme.spacing / 2 + "px " + theme.spacing + "px",
        color: contiamo_ui_utils_1.readableTextColor(backgroundColor)(["black", "white"])
    };
});
var Header = function (_a) {
    var className = _a.className, css = _a.css, children = _a.children, color = _a.color;
    return (React.createElement(Container, { css: css, className: className, color: color }, children));
};
exports.default = Header;
//# sourceMappingURL=Header.js.map