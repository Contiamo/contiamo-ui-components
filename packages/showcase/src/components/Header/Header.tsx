import * as React from "react"
import { Route } from "react-router-dom"

import { Header } from "@operational/components"

const AppHeader: React.SFC<{}> = () => (
  <Header
    css={{
      boxShadow: "0px 1px 2px #d3d1d1",
      backgroundColor: "#fff"
    }}
    color="#fff"
  >
    Contiamo UI
    <Route
      path="/components"
      render={() => (
        <div>
          &nbsp;/&nbsp;<strong>Components</strong>
        </div>
      )}
    />
  </Header>
)

export default AppHeader
