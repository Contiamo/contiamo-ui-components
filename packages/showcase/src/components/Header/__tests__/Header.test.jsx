import React from 'react'
import { shallow } from 'enzyme'

import Header from '../Header'

describe('App Showcase: Header', () => {
  it('Should render correctly', () => {
    expect(shallow(<Header />)).toMatchSnapshot()
  })
})
