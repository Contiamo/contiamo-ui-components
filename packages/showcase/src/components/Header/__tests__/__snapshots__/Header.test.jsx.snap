// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Showcase: Header Should render correctly 1`] = `
<glamorous(Header)
  color="#fff"
  css={
    Object {
      "backgroundColor": "#fff",
      "boxShadow": "0px 1px 2px #d3d1d1",
    }
  }
>
  Contiamo UI
  <Route
    path="/components"
    render={[Function]}
  />
</glamorous(Header)>
`;
